File: source-connector/src/main/java/sourceconnector/domain/processor/AbstractFilterProcessor.java
package sourceconnector.domain.processor;

public abstract class AbstractFilterProcessor<Log> extends BaseProcessor<Log> {
  protected abstract boolean condition(Log input);

  @Override
  public Log process(Log input) {
    if(!this.condition(input)) return null;
    if(this.next == null) return input;
    return this.next.process(input);
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/processor/AbstractFilterProcessor.kt
package sourceconnector.domain.processor

abstract class AbstractFilterProcessor<Log> : BaseProcessor<Log>() {
  protected abstract fun condition(input: Log): Boolean

  override fun process(input: Log): Log? {
    if (!this.condition(input)) return null
    if (this.next == null) return input
    return this.next!!.process(input)
  }
}
